### PR TITLE
[WIP] Add Python 3 support

### PR DESCRIPTION
**Pull request by redapple**

### CODE REVIEW

File: scrapyd-client/scrapyd-deploy [hunk 1]
```diff
@@ -218,24 +223,24 @@ def _add_auth_header(request, target):
...
         _log("Server response (%s):" % f.code)
-        print f.read()
+        print(f.read())
         return True
-    except urllib2.HTTPError, e:
+    except HTTPError as e:
         _log("Deploy failed (%s):" % e.code)
         resp = e.read()
         try:
             d = json.loads(resp)
```
Comment (kmike) [inline scrapyd-client/scrapyd-deploy]: I'm not sure it works because it seems `resp` will be a bytes object, while in Python 3 json requires str.

Comment (redapple) [inline scrapyd-client/scrapyd-deploy]: Right, I need to test the failure cases.

**Comment by podolskyi** [comment]
Any update? maybe you can create a new branch with support python3 but not production ready?
similar scrapyd (python3-wip)

**Comment by rolele** [comment]
I like the idea about creating a branch python3-wip on this project.
Please, can you do that?

**Comment by Digenis** [comment]
@redapple, I can merge this in a new scrapyd branch called python3-wip
and open a new PR from scrapyd/python3-wip to scrapyd/master
since users seem to feel more comfortable
if they use the scrapyd remote to install from git.
I think we'd benefit from them testing it.
Are you ok with this?

**Comment by podolskyi** [comment]
@Digenis @redapple Thanks guys.

**Comment by Digenis** [comment]
Continued at #29.
@rolele, @podolskyi you can try it with
`pip install git+https://github.com/scrapy/scrapyd-client.git@python3-wip`

**Comment by podolskyi** [comment]
@Digenis  ok, will soon try to use it.

**Comment by Digenis** [comment]
We'll be very happy to receive any feedback,
on the python3 ports of both scrapyd and scrapyd-client.

**Comment by podolskyi** [comment]
sure, I will be used py3 version in my new project.

**Comment by rolele** [comment]
Same for me, I will be working with the py3 version. Thanks.

**Pull request merged**